### PR TITLE
`Plagiarism checks`: Do not show the plagiarism case button if students have not yet been notified

### PR DESCRIPTION
**Pull request by krusche**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added `@PreAuthorize` and checked the course groups for all new REST Calls (security).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context
Some students saw a button `Plagiarism Case` in the exercise details even they have not been notified yet.

### Description
This PR also improves the following:
* Validation of certain REST calls related to plagiarism checks and plagiarism cases
* Reduces the amount of unnecessary data sent to the client
* Improves the performance of some actions
* Improves the usability of the plagiarism case detail page (e.g. show the student notification immediately after posting it)
* Makes sure (client side) that instructors only determine the verdict after having notified the student

Thanks to @akesfeden for helping out in the implementation 👍 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Text Exercise

You basically need to check the whole plagiarism process (as described in the documentation)
1. Submit two larger identical texts with student1 and student2
2. Instructor: check for plagiarism (in course administration exercise details)
3. Student1 and student2: verify that the button `Plagiarism Case` is **not** shown in exercise details 
4. Instructor: notify the students by saving the post (you can keep the default text). 
5. Student1 and student2: verify that the button `Plagiarism Case` is now shown in exercise details
6. Student1 and student2: check that you can click on the button and see the anonymized comparison, then answer the post in both cases
7. Instructor: Set two different verdicts for the two cases: once plagiarism and once point deduction
8. Instructor: Check the course scores and make sure both decisions are reflected here accordingly (note: this was not affected in this PR, but should still work)
9. Student1 and student2: check that you can see the corresponding verdict

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
no actual changes

### Screenshots
no actual changes

### CODE REVIEW

File: src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts [hunk 1]
```diff
@@ -92,7 +92,7 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
     hasSubmissionPolicy: boolean;
     submissionPolicy: SubmissionPolicy;
     exampleSolutionCollapsed: boolean;
-    plagiarismCase?: PlagiarismCase;
+    plagiarismCaseId?: number;
```
Comment (Strohgelaender) [inline src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts:95]: unused import in this file.

File: src/main/java/de/tum/in/www1/artemis/repository/plagiarism/PlagiarismCaseRepository.java [hunk 2]
```diff
@@ -37,11 +37,12 @@ public interface PlagiarismCaseRepository extends JpaRepository<PlagiarismCase,
 
     @Query("""
             SELECT DISTINCT plagiarismCase FROM PlagiarismCase plagiarismCase
+            LEFT JOIN FETCH plagiarismCase.post
             WHERE plagiarismCase.exercise.id = :exerciseId
             AND plagiarismCase.student.id = :userId
-            AND (plagiarismCase.post IS NOT NULL OR plagiarismCase.verdict IS NOT NULL)
+            AND (plagiarismCase.post IS NOT NULL)
```
[the remark this answers is not in the document]
Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/repository/plagiarism/PlagiarismCaseRepository.java]: The query was not working, I had to change it anway

**Review by Santia-go — COMMENTED** [review]
Tested on TS1
I tested with student 1 and 3. 
Without notifying anyone, no plagiarism case button is shown. However, when I notify student 3, the button for this student is available, but also for student 1 (I didn't notify this student).
When I press the plagiarism case button as student 1, I can see the comparison, but no the message, because (obvious) I didn't notify to student 1. 

When I notify student 3, I see, as student 1 the button: 
Screenshot 2022-07-31 at 21 31 25

When I press the button as student 1 I see the comparison: 

Screenshot 2022-07-31 at 21 37 40

File: src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts (hunk 1 above)
Comment (krusche) [reply src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts:94]: I fixed it

**Comment by krusche** [comment]
@Santia-go thanks for the feedback. I just tested exactly this case with 3 students and only the ones who have been notified, can actually see the button in my case.
There was an issue in a version before, that I resolved in my commit 607477b6ee13991d915269e6782cf5cfad104088 at 21:27. Maybe, you tested an outdated version. Could you please test again?

**Review by akesfeden — DISMISSED** [review]
Tested it locally, checking for plagiarism, notifying students, settings verdicts works as expected. I couldn't reproduce the issue mentioned by @Santia-go, in my case only the notified student is able to see the "Plagiarism Case" button, is it possible that the unnotified student has another role (tutor, editor, etc.) in the course in your case?

**Review by joaquinmurcia — APPROVED** [review]
Tested on TS1 with the last changes. Indeed, only the students who have been notified can see the plagiarism case notification. Everything works as expected. Some screenshots
1. Confirmed plagiarism case, notification sent to the other student:
Screenshot from 2022-07-31 23-19-22
2. After sending notification to this student:
Screenshot from 2022-07-31 23-20-44

**Review by akesfeden — APPROVED** [review]
Re-approving after the typo fix in server test, the changed file ran successfully for me locally.

**Review by joaquinmurcia — APPROVED** [review]
Re-approving after a typo fix.

**Pull request merged**